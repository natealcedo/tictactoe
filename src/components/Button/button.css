.button-row {
  margin: 0 auto;
}

.button {
  border: 1px solid #c0c0c0;
  height: 50px;
  width: 300px;
  text-align: center;
  border-radius: 5px;
  background-color: #00363d;
  color: white;
  display: flex;
  margin: auto;
  margin-bottom: 5px;
}

.button:hover {
  cursor: pointer;
}

.button > div {
  margin: auto;
}