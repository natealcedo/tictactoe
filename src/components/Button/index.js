import React from "react";
import propTypes from "prop-types";
import "./button.css";

const Button = ({ label, onClick }) => (
  <div className="button" onClick={() => onClick()}>
    <div>{label}</div>
  </div>
);

Button.propTypes = {
  label: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired
};

export default Button;
