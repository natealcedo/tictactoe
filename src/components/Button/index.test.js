import React from "react";
import Button from ".";
import { shallow } from "enzyme";

describe("Button", () => {
  test("Should render without exploding", () => {
    const wrapper = shallow(<Button label={"Button"} onClick={() => {}} />);
    expect(wrapper)
      .toIncludeText("Button")
      .toMatchSnapshot();
  });

  test("Should call onClick when clicked", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button label={"Button"} onClick={onClick} />);
    wrapper.simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
});
