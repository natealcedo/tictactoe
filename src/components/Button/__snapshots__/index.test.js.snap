// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render without exploding 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  <div>
    Button
  </div>
</div>
`;
